File: server/app/models/agent.py
from sqlalchemy import Column, Integer, String, Text, DateTime
from sqlalchemy.orm import relationship
from datetime import datetime
from app.database import Base
from .base import LOCALTIME


class Agent(Base):
    """
    Agent model representing an AI assistant configuration.
    
    An agent defines the behavior and capabilities of an AI assistant,
    including its character settings (personality, style, identity)
    and the LLM/embedding configurations to use.
    """
    __tablename__ = "agents"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(255), nullable=False)
    character_settings = Column(Text, nullable=False, default='')  # Agent's personality, style, identity
    llm_config_id = Column(Integer, nullable=False, index=True)
    embedding_config_id = Column(Integer, nullable=False, default=0, index=True)
    description = Column(Text, nullable=False, default='')
    avatar = Column(String(500), nullable=False, default='')  # Relative path under PrivateBuddyData/avatars/
    created_at = Column(DateTime(timezone=True), default=datetime.now, server_default=LOCALTIME, nullable=False)
    updated_at = Column(DateTime(timezone=True), default=datetime.now, server_default=LOCALTIME, onupdate=datetime.now, nullable=False)

    llm_config = relationship(
        "LLMConfig",
        back_populates="agents",
        primaryjoin="Agent.llm_config_id == LLMConfig.id",
        foreign_keys="Agent.llm_config_id"
    )

File: server/internal/model/agent.go
package model

import "time"

// Agent represents an AI assistant configuration.
//
// An agent defines the behavior and capabilities of an AI assistant,
// including its character settings (personality, style, identity)
// and the LLM/embedding configurations to use.
type Agent struct {
	ID                int64     `gorm:"primaryKey;autoIncrement" json:"id"`
	Name              string    `gorm:"type:varchar(255);not null" json:"name"`
	CharacterSettings string    `gorm:"type:text;not null;default:'';column:character_settings" json:"character_settings"` // Agent's personality, style, identity
	LLMConfigID       int64     `gorm:"not null;index;column:llm_config_id" json:"llm_config_id"`
	EmbeddingConfigID int64     `gorm:"not null;default:0;index;column:embedding_config_id" json:"embedding_config_id"`
	Description       string    `gorm:"type:text;not null;default:''" json:"description"`
	Avatar            string    `gorm:"type:varchar(500);not null;default:''" json:"avatar"` // Relative path under PrivateBuddyData/avatars/
	KnowledgeBaseIDs  string    `gorm:"type:text;not null;default:'[]'" json:"knowledge_base_ids"` // JSON array of knowledge base IDs
	CreatedAt         time.Time `gorm:"not null;autoCreateTime" json:"created_at"`
	UpdatedAt         time.Time `gorm:"not null;autoUpdateTime" json:"updated_at"`
}

func (Agent) TableName() string { return "agents" }
